mypy/messages.py

```diff
@@ -30,7 +30,7 @@
 INVALID_EXCEPTION = 'Exception must be derived from BaseException'
 INVALID_EXCEPTION_TYPE = 'Exception type must be derived from BaseException'
 INVALID_RETURN_TYPE_FOR_YIELD = \
-    'Iterator function return type expected for "yield"'
+    'Generator or supertype function return type expected for "yield"'
```

Comment:
I don't see any clear way of setting the error message line to the function, though...